Complete: Assessment.md and README.md

huangs1 opened the pull request:
The Assessment team has worked together to complete the `assessment.md` fully and have added the relevant information in the `README.md`.

Please provide any feedback relating to formatting and content to the below who will pass it to appropriate members for revision:

* @ilikerustoo
* @huangs1
* @ohnoanarrow
* @yeej2
* @Lancasterwu

Lancasterwu committed:
Update assessment.md

Commit:
Requirements Engineering Scale Added

everitt-andrew committed:
Half of professional skills assessment form completed

Lancasterwu committed:
Finished Interaction with the teaching assistants

Lancasterwu committed:
Fixed the line length limitation

everitt-andrew committed:
updated formatting to match other groups

Commit:
This is my first commit

Alex-Yarkosky committed:
added basic framework for bulletpoints I am responsible for and gave details for 2 of them

Commit:
Added assessment scale for Software Architecture section

Commit:
My first commit for my branch

Commit:
Foundation of software engineering practices: small grammer update

Commit:
indentation

Commit:
indentation

mmarconi committed:
Update assessment.md

ALEXANDERB82 committed:
Foundations of software engineering practices

small grammar change

huangs1 committed:
Update assessment.md

leonardoz15 committed:
Merge pull request #1 from everitt-andrew/ALEXANDERB82-patch-1

Foundations of software engineering practices

everitt-andrew committed:
Another semantic update

libailong committed:
Update assessment.md

Commit:
Managing Software Complexity and Mitigating Software risk worked on

mmarconi committed:
Update assessment.md

everitt-andrew committed:
Two letter changes

Commit:
Software Design to assessment sheet

Commit:
partially filled out 'rubric' for github flow, issue tracker, and commit messages

everitt-andrew committed:
Fixed merge error

everitt-andrew committed:
Fixed spacing

Commit:
added assessments

Commit:
Added guidlines for creating and using repo branches

everitt-andrew committed:
More spacing adjustments

libailong committed:
Update assessment.md

line up

huangs1 committed:
Fixed style guides

huangs1 committed:
updated professional skills for style guide

huangs1 committed:
Merge pull request #2 from huangs1/Interaction-with-instructor-and-ta

Interaction with instructor and ta

huangs1 committed:
Merge pull request #4 from huangs1/customerandteamleaders

Customer and Team Leaders from professional

libailong committed:
Update assessment.md

remove space

libailong committed:
Update assessment.md

change line length

libailong committed:
Update assessment.md

remove spaces

libailong committed:
Update assessment.md

Alex-Yarkosky committed:
added third grading scale that I was responsible for

huangs1 committed:
Updated the format to match the mini-group

mmarconi committed:
Update assessment.md

Lancasterwu committed:
Update assessment.md

huangs1 committed:
Merge pull request #3 from huangs1/first_four_sections_of_Group

first four sections of group

huangs1 committed:
Merge pull request #5 from huangs1/libailong-patch-1

Second half of the group section

Lancasterwu committed:
Update assessment.md

Lancasterwu committed:
Update assessment.md

Lancasterwu committed:
Update assessment.md

Lancasterwu committed:
Update assessment.md

Lancasterwu committed:
Update assessment.md

huangs1 committed:
Merge pull request #6 from everitt-andrew/master

 Merge "Individual" & "revision of guides" to the professional skill category

Commit:
Completed Foundations of Software Engineering part of assessment sheet

huangs1 committed:
Fixed contractions and some grammar errors

Commit:
Wrote the rubric for the last 9 to 11 requirements in Software Development with Python

paladp committed:
Finished first four items

yeej2 committed:
Merge pull request #1 from yeej2/palad-first-4

Palad first 4

yeej2 committed:
Merge pull request #2 from yeej2/yeej2-second-4

added assessments

Commit:
Added in guidlines for utilizing forks

Commit:
filled out assessment sheet items: issue tracker, commit messages and github flow

Commit:
My part completed

Commit:
Committing the work done on the assessment sheet

Commit:
Merge pull request #1 from ohnoanarrow/alex-travis

Alex travis

Commit:
Merge pull request #2 from ohnoanarrow/PaulGuidlines

Paul guidlines

Commit:
Merge pull request #3 from ohnoanarrow/tara-travis

Committing the work done on the assessment sheet

Commit:
Merge branch 'master' into finneyj

Commit:
Merge pull request #4 from ohnoanarrow/finneyj

Finneyj

Commit:
Fully completed the last 3 questions within the 'Software Development with Python' subsection. Fixed mdl markdown issues to pass Travis CI check

ilikerustoo committed:
Merge pull request #3 from yeej2/khanm-last-3

Khanm last 3

baldeosinghm committed:
Corrected format of Software Development with Python

baldeosinghm committed:
Corrected paragraph structure of assessment.md

baldeosinghm committed:
Merge branch 'master' into baldeosingh

baldeosinghm committed:
Restructured assessment.md, again...

baldeosinghm committed:
Merge pull request #4 from yeej2/baldeosingh

Baldeosingh

baldeosinghm committed:
Revert "Baldeosingh"

baldeosinghm committed:
Merge pull request #5 from yeej2/revert-4-baldeosingh

Revert "Baldeosingh"

Commit:
Adjusting line spacing

baldeosinghm committed:
Last commit. Assignment finished.

baldeosinghm committed:
Merge branch 'master' into baldeosingh

baldeosinghm committed:
Merge pull request #6 from yeej2/baldeosingh

Last commit. Assignment finished.

Lancasterwu committed:
Started Updating README

Lancasterwu committed:
Updated a little bit more

Lancasterwu committed:
Merge pull request #1 from Lancasterwu/patch-2

Updated a little bit more

huangs1 committed:
Updated the formatting for Professional Skills

huangs1 committed:
Updated formatting for professional skills

huangs1 committed:
Updated professional skills for format to pass CI

huangs1 committed:
Fixed a grammar error

Lancasterwu committed:
Update README.md

Lancasterwu committed:
Update README.md

Lancasterwu committed:
Update README.md

yeej2 committed:
Fixed formatting for Travis

yeej2 committed:
Updated Assessment Condition

Lancasterwu committed:
Update README.md

yeej2 committed:
Fixed Trailing Spaces

Lancasterwu committed:
Update README.md

Lancasterwu committed:
Update README.md

szklenskij committed:
Testing for Travis

Commit:
Committing the reformatted assessment file

Commit:
Merge pull request #5 from ohnoanarrow/tara-travis

Committing the reformatted assessment file

Commit:
Fixed errors for mdl and proselint

livingstonp729 committed:
Update README.md

wattob committed:
Foundations of Software Engineering to README

huangs1 committed:
Merge pull request #8 from yeej2/master

Section Complete: Software Development with Python

Lancasterwu committed:
Merge pull request #4 from wattob/patch-3

Foundations of Software Engineering to README

leonardoz15 committed:
Added Row #3's work description

Lancasterwu committed:
Update README.md

Lancasterwu committed:
Merge pull request #2 from livingstonp729/patch-2

Updates to the challenges section

leonardoz15 committed:
Merge branch 'patch-1' into patch-2

huangs1 committed:
Merge pull request #10 from szklenskij/master

Section Complete: Foundations of Software Engineering

Lancasterwu committed:
Merge branch 'master' into patch-1

Lancasterwu committed:
Merge pull request #6 from Lancasterwu/patch-1

Patch 1

Lancasterwu committed:
Merge pull request #5 from leonardoz15/patch-2

Added Row #3's work description

Commit:
Merge branch 'master' into master

Lancasterwu committed:
Merge pull request #7 from Lancasterwu/patch-1

Patch 1

Commit:
Merge pull request #9 from ohnoanarrow/master

Completed sections for Project Management and Continuous Integration

Lancasterwu committed:
Update README.md

huangs1 committed:
Fixed Spacing for Styleguide

huangs1 committed:
Updated for spacing for line 404

Lancasterwu committed:
Update README.md

huangs1 committed:
Fixed spacing for project management section

Lancasterwu committed:
Update README.md

Lancasterwu committed:
Update README.md

Lancasterwu committed:
Update README.md

huangs1 committed:
Updated Conduct for consistency

finneyj2 committed:
Update README.md

Lancasterwu committed:
Merge pull request #9 from finneyj2/patch-3

Update README.md

Lancasterwu committed:
Update README.md

Lancasterwu committed:
Update README.md

Lancasterwu committed:
Update README.md

Lancasterwu committed:
Update README.md

Lancasterwu committed:
Update README.md

Lancasterwu committed:
Update README.md

huangs1 committed:
Merge pull request #11 from Lancasterwu/master

README Completed

Michionlion commented on README.md:
```diff
@@ -9,11 +9,12 @@ skills listed on this assignment sheet.
 
 ## Improving the Assessment
 
```
This header no longer makes sense with the body of the section. This section could perhaps be changed to reflect the complete nature of the document. Additionally, there should be an easy to see link to the actual document in the readme, which can be accomplished using something like: `[document](assessment.md)`.

paladp replied:
Handling this right now

yeej2 replied:
This has been resolved by paladp

Michionlion commented on README.md:
```diff
@@ -22,12 +23,82 @@ assessment at one of the following levels:
-professional skills will result in an overall assessment for a software project.
-Once you are finished with the assessment, please update the README as
-necessary.
+## How we design Assessment Sheet
```
I'm concerned with the relevance of the below section. It could perhaps be shorter, or more concise. Perhaps some of the assigned members could provide some justification? Additionally, users should be referenced with `@` so as to take advantage of GitHub's user linking feature.

Lastly, there are some grammatical and other writing mistakes in this section, which should have been caught by `mdl` or `proselint`, such as mixed tenses and the concatenation of "to merge" on line `32`. I would highly suggest an additional proof-reading session by a few team members.

yeej2 replied:
This has been resolved by paladp

Michionlion commented on README.md:
```diff
@@ -22,12 +23,82 @@ assessment at one of the following levels:
+* ohnoanarrow (Tara Douglass) - Setup and configuring Tracis CI, performing secure
+  releases of project deliverables, and third party integration with Travis CI
+
+## Challenges When Making Assessment Sheet
```
Although relevant to this project as a lab assignment, this section is not relevant to the outside observers who are part of your audience. There should be a compelling reason for it's inclusion.

sutterj replied:
This all looks like stuff that should be a part of your individual reflection document in your personal repos rather than in the README.

yeej2 replied:
This has been resolved by paladp

Michionlion commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
 ### Software Development with Python
```
There is no description of the actual use of these levels. Some mention should be made of students being assessed based on these levels of adequacy, introducing this document.

paladp replied:
Handling this right now

paladp replied:
@Michionlion This should be resolved in commit #22 on spencer's fork

paladp replied:
@Michionlion This should be resolved in commit #22 on spencer's fork

Michionlion commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
+  functionalities of `pylint` and `flake8` such as suppressing messages and
+  overrides
+  * E = Same requirements as level 'G' in addition to being able to help other
+  students trouble shoot `pylint` and `flake8`
```
`troubleshoot` instead of `trouble shoot` is the correct use here, as well as being the standard in this document.

paladp replied:
Also handling this right now

paladp replied:
@Michionlion This should be resolved in commit #22 on spencer's fork

paladp replied:
@Michionlion This should be resolved in commit #22 on spencer's fork

Michionlion commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
+  * N = `black` not installed or setup
+  * I = Unable to run `black` to format Python programs
+  * A = Able to run `black` to format Python programs
+  * G = Same requirements as level 'A' in addition to solving
```
This requirement of solving problems with the installation does not make too much sense to me. What about students who don't encounter any problems? Do they not meet the requirements for 'Good' work? Unless there is a compelling reason, I would suggest this section be reworked a little.

paladp replied:
Handling this right now as well

paladp replied:
@Michionlion This should be resolved in commit #22 on spencer's fork

Michionlion commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
+  * I = Found many errors that were not resolved in testing
+  * A = Found errors that were not resolved
+  * G = Found few errors that were not resolved
+  * E = Found minimal errors and resolved most errors
```
These sections are about **testing** with `pytest`, correct? Not about the correctness of code. It should be based on the competency of `pytest` use, such as the ability to write test cases.

yeej2 replied:
Handling this issue

yeej2 replied:
Finished this issue

Michionlion commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
+  * G = Found few errors that were not resolved
+  * E = Found minimal errors and resolved most errors
 * Calculating code coverage of a `pytest` test suite
+  * N = Did not utilize `pytest` for code coverage
```
Again with these sections, I would almost suggest either a (Y)es or (N)o level of assessment, because the level of code coverage has *nothing* to do with calculating or reporting said code coverage. There is a distinction between being able to do a certain task at some level of competency, and the output produced by that task.

yeej2 replied:
Handling this issue

yeej2 replied:
Finished this issue

Michionlion commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
+  * E = Student successfully added docstring with useful description and proper
+  syntax
 * Adopting and applying appropriate naming conventions
+  * N = Student used inappropriate naming styles that are neither prescriptive
```
It is not clear in these sections what correct syntax is. A style guide or mention of one should be made.

Additionally, "Students" or "Student" should not be used as an introductory word. Instead, the sentence should simply begin with the requirement.

baldeosinghm replied:
Now working on this

baldeosinghm replied:
Finished

Michionlion commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
 * Using exception handling to create a robust Python program
+  * N = Student does use exception handling for the possibility of unexpected
+  user input
+  * I = Student introduces try and except statements but does not have any clause
```
The distinction of a "clause" executing or not does not make much sense -- does "not execute" mean that it crashes? Or that not all exceptions are handled? This section is unclear and should be reworked.

baldeosinghm replied:
Will be working on this as well

ilikerustoo replied:
This has been revolved and merged

Michionlion commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
+  * G = Student correctly codes exception handling, but clause doesn't fully execute
+  * E = Student correctly codes exception handling with clause being perfectly executed
 * Refactoring a Python program to improve its characteristics
+  * N = No refactoring done or changed functions completely
```
Level 'N' in this section makes no sense. Refactoring can sometimes involve completely changing functions, or rewriting entire modules. Additionally, refactoring is not the same as making maintainable, which level 'A' seems to suggest.

ilikerustoo replied:
Changed to 
 * N = No refactoring done
  * I = Little code refactored and no documentation
  * A = Some code refactored and changes documented
  * G = Refactored code contains easy to understand functions and documentation
  * E = Refactored code is clean, efficient, and maintainable code that is well documented

Michionlion commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
+  * G = Includes easy to understand functions and documentation
+  * E = Clean and efficient code that is well documented
 * Debugging Python programs using logging and other methods
+  * N = No debugging done
```
This is only a suggestion (at a much lower level of severity than other comments here), but it may be useful to separate "bugfixing" from "bugfinding".

ilikerustoo replied:
Changed to
* N = No bigfixing or bugfinding done
  * I = Identified bugs in code but no fix
  * A = Fixed some identified bugs using logging/other methods
  * G = Fixed most identified bugs using logging/other methods
  * E = Fixed all bugs using logging/other methods and well documented

Michionlion commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
+  * I = Language constructs used inappropriate way
+  * A = Some language constructs used appropriately
+  * G = Some or many language constructs used efficiently
+  * E = Several language constructs used efficiently and reason is documented
```
There is not an easy distinction between levels 'G' and 'E' here. Is "many" less than "several"? Or is the documentation part more important? More emphasis should be made.

ilikerustoo replied:
Changed to
* N = No written code or monolithic code
  * I = Language constructs used inappropriately
  * A = Some language constructs used appropriately
  * G = Most language constructs used efficiently
  * E = All language constructs used efficiently and reason is documented

Michionlion commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
 ### Project Management with GitHub
 
 * Using the issue tracker
+  * N = None would indicate that someone did not use the issue tracker.
```
In all of the below sections, sentences should be reworked so that they follow the descriptive style found in other sections -- "Used good syntax", etc. "Good would indicate" is redundant.

For example, line `126` should read `  * N = No use of the issue tracker`.
Additionally, periods are used at the end of these lines, contrary to your internal style.

ilikerustoo replied:
This is probably @finneyj2 section's. Although I am unsure if she has looked at this at all. I will attempt to fix this in the meantime.

ilikerustoo replied:
I have made the changes and merged

toccinAC commented:
I do not know if the information in the README is necessary. It is meant to explain what the guide is, not how it was made specifically be certain people. Unless @gkapfham wants this information in this document.

Michionlion commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
+  correctly. Additionally, merges and pull requests are done correctly and the
+  commit messages can be understood, but the programmer pushed to the master
+  branch.
+  * G = Good would indicate that the issue posted is neither redundant nor
```
This seems to be a level 'G' for the issue tracker section, not GitHub flow.

huangs1 replied:
@ilikerustoo is working on this issue

ilikerustoo replied:
This is @finneyj2's section. This section has been re-written by me to fix the issue.

Michionlion commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
+  label. Had strong commit messages, in addition to using and creating
+  repository branches, forks, and pull request correctly.
 * Creating and using a repository branch
+  * N = Routinely committing edits to the master branch or not pushing to
```
Appending "will receive a __ grade" is redundant.

Lancasterwu replied:
Fixed by https://github.com/huangs1/assessment-professional-skills/pull/25/files

Michionlion commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
+  adequately describe the branch's purpose receive an adequate grade
+  * G = Well documented changes to correct well-named and necessary branches
+  will receive a good grade
+  * E = Immaculately documented changes on branches that serve extremely
```
The distinction of necessary branches to extremely important branches does not make sense to use as a grade level. Do students who work on a large number of small, unimportant branches receive a lower grade?

yeej2 replied:
I will fix this issue

yeej2 replied:
I have fixed this issue

Michionlion commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
+  * E = Immaculately detailed code and communication with the team before
+  merging receives an excellent grade
 * Creating and reviewing a pull request
+  * N = no PR was made
```
Incorrect capitalization.

In this section, mentioning "A PR was made" is redundant.

huangs1 replied:
This issue should have been resolved by this pull request: https://github.com/huangs1/assessment-professional-skills/pull/14

Michionlion commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
+  fully discusses the ramifications of the PR after successfully building with
+  Travis CI
 * Using appropriate commit messages
+  * N = None would indicate that the commit message was nonsensical. An example
```
Starting a sentence with "Adequate", etc. is redundant. Use prescriptive style sentences similar to other sections in this document. In addition, an example commit message should not mention something is broken in the main tagline -- that is for an issue! If something was fixed, then the message should read something like `Fix main method of X.py by ___`.

Michionlion commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
 ### Continuous Integration with Travis
 
 * Setup and configure Travis CI
+  * N = Fails to perform any setup and configuration of Travis CI
```
This is admittedly nitpicky... but the beginning tenses here are different from the rest of the document.

huangs1 replied:
Pull request #26 on https://github.com/huangs1/assessment-professional-skills/pulls should fix the issue

Michionlion commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
 * Requirements engineering
+  * N = No effort to document, define, or maintain requirements present
+  * I = Brief or unspecific documentation, definition, and maintenance given
+  * A = Explains what was done with enough detail to convey the most important information
```
Level 'A' does not seem to describe requirements engineering -- are you documenting work done, or maintaining requirements? It is not clear.

huangs1 replied:
This issue should have been resolved by this pull request: https://github.com/huangs1/assessment-professional-skills/pull/23

Michionlion commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
+  * E = The software engineer displayed highly detailed, elaborate, and complete
+  documentation, definition, and maintenance in the given design process
 * Software architecture
+  * N = There is no concrete structure and the master branch is broken
```
Software architecture should not rely on software implementation. This applies to both below sections. @sutterj or @toccinAC or @gkapfham feel free to correct me on this one.

Michionlion commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
+  * A = Documentation has a bit of detail for every main part of the program
+  * G = Documentation is detailed and visible for most of the software within the
+  source code
+  * E = Explains in great detail how to use and operate the computer software
```
Is documentation only about how to use and operate the software? What about the constraints, why a certain approach was chosen, justification, description of individual methods, etc.?

Michionlion commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
+  * E = Everything is exceedingly well organized, with no excess or lack of files,
+  folders, branches, forks, etc.
 * Mitigating software risk
+  * N = No procedure is followed, the master branch is broken, or files are corrupted
```
I'm not sure this section correctly identifies good mitigation techniques, only the outcomes. This may work, but identifying techniques is probably fairer. A good, short guide on the subject I found is [here](https://www.test-institute.org/What_Is_Software_Risk_And_Software_Risk_Management.php). Of course, @gkapfham, @toccinAC, and @sutterj, feel free to correct me.

huangs1 replied:
Pull Request https://github.com/huangs1/assessment-professional-skills/pull/20 should have fixed this issue

Michionlion commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
+        to each project
+    * G = An above average amount of knowledge and contributions
+        to each project
+    * E = Going above and beyond the expectations and contributions
```
It is my view that an assessment level of 'E' should be possible for everyone. Depending on going "above and beyond" others removes that. This is of course entirely personal, and so I would not suggesting changing it unless you have no reason for the current state.

sutterj replied:
I actually do think that it should be changed because all levels should be able to be obtained by all individuals.  Work that is beyond the required scope is great and strongly encouraged, but I do not think that it should be part of the requirements.

Michionlion commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
+        instances of overriding the master branch
+    * I = Occasional disregard for the model including overriding
+        the master branch
+    * A = Almost always adhering to model, but never overriding
```
What is "overriding the master branch"? Correct terminology should be used.

leonardoz15 replied:
This was fixed by my pull request [#15](https://github.com/huangs1/assessment-professional-skills/pull/15)

Michionlion commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
+    * E = Working well with a team all time with clear and constructive
+      communication
 * Understanding and avoiding red flags
+  * N = Ignoring all red flags
```
Where do you find what is a red flag? Is that important here? This section is confusing.

gkapfham replied:
This is a good point! If you check the "Philosophy of Software Design" book you can see some examples of these red flags. Perhaps some examples are needed in the assessment guide?

leonardoz15 replied:
Me and my row are working on this change.

everitt-andrew replied:
This suggestion has been addressed in a new pull request from @leonardoz15

Michionlion commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
+  * G = Missing two lab sessions
+  * E = Missing one or less lab sessions
 * Deadlines
+  * N = If an assignment is turned in 5 or more days past the deadline
```
Levels that are the same should not be repeated. Use some means of reducing this redundancy.

libailong replied:
I totally agree with that and I will take a look to see what can I revise

Michionlion commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
+  * G = Regularly handles differences with teammates in mature manner
+  * E = Always handles differences with teammates in mature manner
 * Infractions
+  * N = Always breaks the team rules
```
This section may not be needed when comparing with the conduct repository's work.

Michionlion commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
+     TAs when they are trying to help
+  * I = Communicated with the TAs poorly, did not use them well as your
+     resources, did not talk to them
+  * A = Communicated with the TAs well, but relied on them too much, asked
```
"most of the work"? That should be much lower on the grading scale in my opinion.

Lancasterwu replied:
I just changed "most of the work" to "too many works". Does that work?

Michionlion replied:
No, @Lancasterwu, it should be "too much work".

Michionlion reviewed (changes requested):
There are some minor issues with tense changes, incorrect writing styles, and period use, but on the whole there is fairly high-quality writing here.

There are quite a few comments below, each of which should be addressed in some way, but many are ignore-able if the issues behind them can be justified.

If all comments have some action taken (whether justification or changes, even if I don't agree with them), I will approve this PR.

Thanks for all of your hard work!

sutterj commented:
I agree with @toccinAC that the README should be an explanation of what the guide is.  If someone who was not familiar with this were to come across it, they should be able to look at the README and figure out what it is they are looking at.

sutterj commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
+  * A = Basic development environment configured, including proper dependencies
+  and development tools.
+  * G = Additional development tools configured, in addition to minimum required
+  dependencies and developments
```
Should "developments" be development tools instead?  You might then consider rewording it to avoid using development tools twice in the same fragment.

ilikerustoo replied:
This was changed to 'Additional development tools configured and able to provide assistance and troubleshooting to other students regarding their own development environment' probably by @paladp

gkapfham reviewed (changes requested):
Thanks for all of your hard work on these two documents. I want to draw your attention to several remaining issues:

* Several people who have contributed to this PR do not have icons or photographs for their GitHub profiles. Could you please add these so that members of the class know who you are? If you have an overriding issue that would prevent you from adding this icon or picture, please DM me in Slack, okay?
* There are several mistakes in capitalization in the assessment sheet. For instance, "How we design Assessment Sheet" does not capitalize "design". There is also a place where you do not capitalize the word "Python".
* There are also several phrases and headers that do not make sense. For instance, "How we design Assessment Sheet" is not phrase in a way that can be easily understood.
* There are spelling mistakes in the current version of the file, with words like "tomerge" currently still written in an incorrect fashion.
* If you are going to use GitHub user names, I think that it makes sense to use @username, right? Otherwise, It is probably better to use your full name. Did you check to see how @username will render inside of a Markdown document?
* I also with @Michionlion that the breakdown of who did what is not very useful for someone who wants to complete the assessment sheet on a weekly basis. Can you justify the inclusion of this content?
* There are several small punctuation mistakes in the document. For instance, some of the assessment levels have a period at the end of the sentence and some do not. You need to be consistent in this regard.
* I also agree with @Michionlion that your content about Pytest is not correct. What you have written is not actually about using the Pytest tool (like we did in the previous practical and read about in the textbook).
* The points that you have raised about debugging are not actually related to the task of debugging.
* The document uses terms like "efficiently" and "effectively" without a clear meaning. What does it mean to use a Python language construct in an "efficient" fashion?
* There is a significant variation in the description of the different levels and the amount of thought that seems to have gone into the description of these levels. Some areas have a phrase, others a sentence, and others have one or more sentences. Also, the capitalization scheme for the points is not consistent.
* The content about deadlines is written in terms of assignments. However, one of the critical points in this class is that the assessment is **not** about assignments and rather about deliverables and features.
* I suggest that instead of using the word "TAs" you instead write "technical leads", which is the term that we will adopt in this course.

I hope that this review helps. If you have questions about these comments, let's discuss them in this PR, okay?

toccinAC commented on README.md:
```diff
@@ -22,12 +23,82 @@ assessment at one of the following levels:
+grading scale(example: the difference between inadequate and good).
+"None" in particular was difficult at times to distinguish from
+"Inadequate" in some cases. But we managed to circumvent these issues
+with communication among personal teams and the larger group.
```
This is information that should be included in the personal reports for each person. The READMe is meant to be a representation of how others should be using the different documents, not the challenges that were faced during the creation.

yeej2 replied:
This has been resolved by @paladp.

toccinAC commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
+  * N = No debugging done
+  * I = Identified bugs in system but no fix
+  * A = Fixed some identified bugs using logging/other methods
+  * G = Fixed most identified bugs using logging/other methods
```
There is no change from the A to the G category here. There should be a steady increase in the ability of each person.

Michionlion replied:
I would disagree here, as 'A' only requires some fixes, whereas 'G' requires most. Perhaps changing to 'few' and 'many' would be better?

ilikerustoo replied:
This has been addressed and changed

toccinAC commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
+  * E = Immaculately documented changes on branches that serve extremely
+  important purposes will receive an excellent grade
 * Creating and using a repository fork
+  * N = Pushing to the wrong fork or making forks that serve no purpose will
```
As @Michionlion has stated previously, there should not be " will receive a __ grade." The indicators, such as "N = ," represents this grade.

ilikerustoo replied:
This has been changed. Thank you

toccinAC commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
+  elements in a complex and complete blueprint of the software system, and the
+  master branch has no errors and contains completed code
 * Software design
+  * N = No problem solving steps taken
```
This section seems to mix the design and the implementation process. There is a difference in these two parts of the process and this should be indicated.

ilikerustoo replied:
This has been changed, thank you

toccinAC commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
+  * G = A software solution is created that includes both an algorithm design and
+  a high-level architecture design
+  * E = All goals and expectations of the design are met and compatibility, reliability,
+  reusability, and scalability are included    while a software solution is ultimately
```
There is unnecessary spacing on line 303.

yeej2 replied:
I will handle this one

yeej2 replied:
I finished this section

toccinAC commented on assessment.md:
```diff
@@ -13,81 +13,512 @@
+  * N = Never participate in group conversation or activity
+  * I = Rarely participate in group conversation or activity
+  * A = Sometimes participate in group conversation or activity
+  * G = Regularly participate in group conversation or activity
```
Is there a difference between regularly and always? This seems confusing.

Michionlion replied:
I agree here. Perhaps using a time amount expected would be better? -- I.E. hours per week spent in communication, or time in between responses, or responsiveness.

yeej2 replied:
I'll handle this one too

yeej2 replied:
I finished this section

Lancasterwu committed:
Updated Interaction with the teaching assistants

Lancasterwu committed:
update line length issue

Lancasterwu commented:
> * I suggest that instead of using the word "TAs" you instead write "technical leads", which is the term that we will adopt in this course.
@gkapfham I just made a change of all use of "TAs" to "technical leads".

Lancasterwu committed:
added "s" to "work" with the use of "too many"

Michionlion commented on assessment.md:
```diff
@@ -489,15 +489,15 @@
-     them to do most of the work
-  * G = Communicated with the TAs well, politely asked them questions and try
+  * A = Communicated with the technical leads well, but relied on them too much, asked
+     them to do too many work
```
"Many work"-> "much work"

Lancasterwu replied:
oh, ok. oops

Lancasterwu committed:
changed many to much

Lancasterwu committed:
pasted something by accident

finneyj2 committed:
Fixed errors in Github flow + commit sections

added lines to commit sections and changed bad examples in github flow

finneyj2 committed:
Travis issues

Attempted to change spacing complaints from Travis

finneyj2 committed:
Inconsistent Indentation Addressed

Alex-Yarkosky committed:
Added missing capitalization for consistency

I believe this proposed change will make all of the first letters of each grading benchmark are capitals (unless unnecessary when the start with terms like `pyenv` or `black` or etc.

finneyj2 committed:
Travis attempt 3

Spacing and identation

wattob committed:
Edits to lines 296 and 303 for Pull Request Standards

leonardoz15 committed:
Updated assessment - red flags

finneyj2 committed:
Indentation and Spacing fix attempt

paladp committed:
Addressed three issues raised. 

Added description for software development with Python section, changed trouble shooting format, changed levels for working with black

leonardoz15 committed:
Updated line length for Travis

finneyj2 committed:
Spacing fixed

Addressed spacing issues

thomad74 committed:
Changes to Mitigating Software Risk section

mmarconi committed:
Changed deadlines and deleted infractions

leonardoz15 committed:
Spacing for Travis

paladp committed:
Addressed issues with formating

szklenskij committed:
Edit to Foundations of Software Engineering

Made appropriate changes to Requirements engineering, Software architecture, Software documentation, and Mitigating software risk sections as suggested by @michionlion

paladp committed:
Removed trailing space

Alex-Yarkosky committed:
Removed redundant language

Redundant language used in the benchmarks for grades that referenced the section that the benchmark is about is no longer there (and should be done for all of the sections that I did where it made sense to do so).

Lancasterwu committed:
Merge pull request #13 from finneyj2/patch-5

Fixed errors in Github flow + commit sections

szklenskij committed:
Edit to Pass Travis

Should have fixed line length and trailing space error

huangs1 committed:
Merge pull request #14 from Alex-Yarkosky/patch-1

Added changes for consistency in sheet

huangs1 committed:
Merge pull request #15 from leonardoz15/patch-1

Updated assessment - red flags

Lancasterwu committed:
Merge pull request #22 from paladp/patch-1

Addressed three issues raised

huangs1 committed:
Edited line length for line 309

huangs1 committed:
Merge pull request #18 from wattob/patch-4

Edits to lines 296 and 303 for Pull Request Standards

huangs1 committed:
Edited line 238 to match tenses

baldeosinghm committed:
Changed exception handling and naming conventions

Improved language of exception handling and naming conventions for more concise description of rubric.

paladp committed:
Addressed wording issue

Lancasterwu committed:
Merge pull request #26 from huangs1/huangs1-setup-and-configure

Edited line 238 to match tenses

Commit:
fixed syntax in lines 50-80 and 300-431

ilikerustoo committed:
Changes to 'Software Development with Python'

Rewording done in the last 3 sections of 'Software Development with Python'

yeej2 committed:
Merge pull request #29 from huangs1/yeej2-patch1

fixed syntax in lines 50-80 and 300-431

huangs1 committed:
Updated mitigating risk section

ilikerustoo committed:
Correct spelling mistake

bigfixing --> bugfixing on line 119

huangs1 committed:
Merge pull request #20 from thomad74/patch-1

Changes to Mitigating Software Risk section

huangs1 committed:
Updated deadlines

huangs1 committed:
Merge pull request #30 from huangs1/ilikerustoo-patch-1

Changes to 'Software Development with Python'

huangs1 committed:
Merge pull request #23 from huangs1/szklenskij-patch-1

Edit to Foundations of Software Engineering

paladp committed:
Initial Change to README

Lancasterwu committed:
Merge pull request #21 from huangs1/mmarconi-patch-1

Changed deadlines and deleted infractions

huangs1 committed:
Merge pull request #27 from paladp/patch-2

Addressed wording issue

paladp committed:
Update README.md

huangs1 committed:
Added in to requirements engineering

paladp committed:
Update README.md

ilikerustoo committed:
Merge branch 'master' into patch-3

Lancasterwu committed:
Merge pull request #31 from paladp/patch-3

Initial Change to README

Lancasterwu committed:
Merge pull request #28 from baldeosinghm/patch-3

Changed exception handling and naming conventions

Michionlion commented on README.md:
```diff
@@ -2,32 +2,33 @@
+* Continuous Integration with Travis
+* Foundations of Software Engineering
+
+The prfessional skills section contains these subsections:
```
`prfessional` -> `professional`

huangs1 committed:
Edited readme for grammar error

Commit:
Fixed syntax

huangs1 committed:
Merge pull request #32 from huangs1/yeej2-patch2

Fixed syntax

ilikerustoo committed:
Rewrote sections in 'Project Management with GitHub'

Rewrote 'Using the issue tracker' and 'Using the GitHub flow model'

ilikerustoo committed:
Fixed CI issues

ilikerustoo committed:
Fixed trailing spaces and line length

assessment.md:120: MD009 Trailing spaces
assessment.md:122: MD009 Trailing spaces
assessment.md:128: MD009 Trailing spaces
assessment.md:130: MD009 Trailing spaces
assessment.md:122: MD013 Line length

Commit:
Fixed syntax

huangs1 committed:
Updated formatting python with 'black'

huangs1 committed:
Merge pull request #33 from huangs1/bugfix/fixing-issueTracker-and-GitHubFlowModel

Rewrote sections in 'Project Management with GitHub'

yeej2 committed:
Merge pull request #35 from huangs1/yeej2-patch2

Fixed syntax

huangs1 committed:
Updated for formatting

Commit:
fixed PR conditions

huangs1 committed:
Merge branch 'master' into updated/formatting-python-with-'black'

huangs1 committed:
Merge pull request #34 from huangs1/updated/formatting-python-with-'black'

Updated formatting python with 'black'

ilikerustoo committed:
Reverted overwritten changes

Fixing Matt's push to master. Reverting changes that were overwritten by his push

huangs1 committed:
Merge pull request #36 from huangs1/revert-mattChanges

Reverted overwritten changes

ilikerustoo committed:
Re-added my changes that were overwritten

Re-added the changes to the 'Software Development with Python' in the last 3 subsections that were overwritten by Josh's changes to master

ilikerustoo committed:
Reverted more changes

Reverted more changes from Josh's push to master

huangs1 committed:
Merge pull request #37 from huangs1/bugfix-Revert-joshChanges

Re-added my changes that were overwritten

Commit:
Fixed PR commits

huangs1 committed:
Merge branch 'master' into bugfix-joshChanges2

huangs1 committed:
Merge pull request #38 from huangs1/bugfix-joshChanges2

Reverted more changes

yeej2 committed:
Merge branch 'master' into fixed-pr-mistakes

Commit:
Fixed proselint

Commit:
Merge branch 'fixed-pr-mistakes' of github.com:huangs1/assessment-professional-skills into fixed-pr-mistakes

huangs1 committed:
Merge pull request #39 from huangs1/fixed-pr-mistakes

Fixed pr mistakes

ilikerustoo committed:
Reverting more work

Updated the pylint/black section in Software Development with Python
Updated Requirements Engineering, Software Architecture, and Software Documentation in Foundations of Software Engineering

ilikerustoo committed:
Fixed trailing space on line 219

assessment.md:219: MD009 Trailing spaces

ilikerustoo committed:
Merge pull request #40 from huangs1/bugfix-moreReversion

Reverting more work

ilikerustoo committed:
Changed Red Flag requirements

Reworded some red flag stuff

ilikerustoo committed:
Fixed trailing space in line 324

assessment.md:324: MD013 Line length

ilikerustoo committed:
Fixes and changes

Adjusted changes in different sections like Software Architecture

Michionlion commented on assessment.md:
```diff
@@ -40,41 +40,31 @@ when creating python programs.
   troubleshooting to other students using `pyenv` and `pipenv`
 * Linting Python programs with `pylint` and `flake8`
-  * N = `pylint` and `flake8` not installed or setup
+  * N = `pylint` and `flake8` not installed,setup, or used
```
Spacing problem between `installed` and `setup`.

Lancasterwu replied:
I think this is resolved and it is showing correctly when I was checking "files changed"

Michionlion commented on assessment.md:
```diff
@@ -12,82 +12,447 @@
+  * N = Impolite and impatient
+  * I = Polite but impatient
+  * A = Polite and patient but does not have clear requests
+  * G = Polite and patient but does not respond to team leaders
```
It may be better to have not responding to team leaders lower on this list...

Lancasterwu replied:
fixed by @ilikerustoo 
https://github.com/Allegheny-Computer-Science-203-S2019/cs203-S2019-assessment/pull/13/commits/58e5df1e075cbcf87e6f0710ebafaadbb0b2584a

ilikerustoo commented:
Oh wow

On Fri, Feb 1, 2019 at 7:30 PM Saejin Mahlau-Heinert <
notifications@github.com> wrote:

> *@Michionlion* commented on this pull request.
> ------------------------------
>
> In assessment.md
> <https://github.com/Allegheny-Computer-Science-203-S2019/cs203-S2019-assessment/pull/13#discussion_r253240416>
> :
>
> >  * Interaction with the team leaders
> +  * N = Impolite and impatient
> +  * I = Polite but impatient
> +  * A = Polite and patient but does not have clear requests
> +  * G = Polite and patient but does not respond to team leaders
>
> It may be better to have not responding to team leaders lower on this
> list...
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Allegheny-Computer-Science-203-S2019/cs203-S2019-assessment/pull/13#pullrequestreview-199319434>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AYVYCygWr2JqLU8koHH24isVCv8I_s9kks5vJNwNgaJpZM4aY_vD>
> .
>

ilikerustoo committed:
Adjusted spacing and adjusted a section

Adjusted "Interaction with the team leaders" and adjusted spacing on line 52

ilikerustoo committed:
Merge pull request #41 from huangs1/bugfix/redFlag-Changes

Changed Red Flag requirements

Lancasterwu commented:
@ilikerustoo @huangs1 Just got a chance to look at this. Is there anything else we need to work on?

Lancasterwu commented on README.md:
```diff
@@ -2,32 +2,33 @@
+
+## Problems with the Assessment Sheet
+
+If you have any problems with this Conduct Guide, then please create an issue
```
Why it is "Conduct Guide"? I am confused. @paladp

Lancasterwu replied:
It is probably fine to just keep it.

Michionlion replied:
@Lancasterwu I would disagree, this seems to be copied from the Conduct Guide's readme, and should be changed to reference the assessment sheet.

Lancasterwu replied:
Ok. I saw it was changed from "Assessment Sheet" to "Conduct Guide" through the "files changed history", so I thought he meant Assessment Sheet is kind of "conduct guide". 
As you mentioned, it is possible for him to copy from Conduct Guide's readme and forgot to change. I think I will just go ahead to change it

Lancasterwu replied:
fixed by https://github.com/huangs1/assessment-professional-skills/commit/250a6cf084a460054a73c704e3275565bce1c1f4

sutterj commented on assessment.md:
```diff
@@ -12,82 +12,447 @@
+     project and course requirements
+  * E = Communicated with the instructor excellently, followed
+     all project and course requirements
+* Interaction with the technical leads
```
There are a couple of issues with tense in this section.

I would suggest the following:

N: ...used hate speed, ignored technical leads...
G: ...and tried to solve...
E: ...politely shared own opinions and worked with them...

Lancasterwu replied:
fixed by https://github.com/huangs1/assessment-professional-skills/pull/43

Lancasterwu replied:
@sutterj Thanks so much for pointing it out!

Lancasterwu committed:
Fixed issues with tense in interaction section

Lancasterwu committed:
Merge pull request #43 from huangs1/fix

Fixed issues with tense in interaction section

Lancasterwu committed:
Edited README. changed last section

"Conduct Guide" to "Assessment Sheet"

Lancasterwu reviewed (dismissed):
I think this assessment sheet is great. I think we can also invite one or two reviewers from the Conduct Guide's team as their representatives.

gkapfham commented:
I should point out that I have removed the request for a review from @toccinAC as he may be busy right now.

I have approved this pull request for a merge to the master branch. Two quick points:

* We are waiting for @paladp to add an icon for his GitHub account
* We need to have @ilikerustoo approve the pull request

After these steps are complete, then I think that the PR can be merged into the main repository!

paladp commented:
Just did it, sorry for the delay!

On Mon, Feb 4, 2019 at 1:21 PM Gregory M. Kapfhammer <
notifications@github.com> wrote:

> I have approved this pull request for a merge to the master branch. Two
> quick points:
>
>    - We are waiting for @paladp <https://github.com/paladp> to add an
>    icon for his GitHub account
>    - We need to have @ilikerustoo <https://github.com/ilikerustoo>
>    approve the pull request
>
> After these steps are complete, then I think that the PR can be merged
> into the main repository!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Allegheny-Computer-Science-203-S2019/cs203-S2019-assessment/pull/13#issuecomment-460355835>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AeM5jOWeplZOTvs_THlunO8vYvpDP5F4ks5vKHowgaJpZM4aY_vD>
> .
>

huangs1 committed:
Improved some sections and formatting

@ilikerustoo @huangs1

ilikerustoo reviewed (changes requested):
We are working on some changes

huangs1 committed:
Made improvements on certain sections

Improved/reworded some sections
Deleted "deadlines, conduct, attendance, interaction with course instructor" We are leaving this up to the conduct team
Me and @ilikerustoo worked on these changes
Changed Revision of Guides to "assessment and Conduct Guide" and made "Revision of Guides" separate section about changes to the guide

huangs1 committed:
Formatting changes to pass CI

Commit:
Formatting changes to pass CI

huangs1 committed:
Merge pull request #44 from huangs1/bugfix/improvement

Improved some sections and formatting

ilikerustoo commented:
Everything is ready to be pushed

Michionlion reviewed (approved):
Everything looks good, much more confident in this version!

Lancasterwu reviewed (approved):
It looks even better now. Thanks for all of your hard work!

Michionlion commented:
@gkapfham Can I get your final approval before this is merged?

ilikerustoo reviewed (approved):
The changes look good

gkapfham reviewed (approved):
Hello @ilikerustoo and @Michionlion, I am writing to formally approve the merge of this PR. I think that we have made a lot of progress on the assessment guide. We might need to tweak the content in the future, but I can try to lead that effort as we start to focus on the GatorGrouper tool.
